tm4c123x-hal/src/timer.rs

```diff
@@ -0,0 +1,166 @@
+                ) -> Self
+                where
+                    T: Into<Hertz>,
+                {
+                    // power up
+                    sysctl::control_power(
+                        pc, sysctl::Domain::$powerDomain,
+                        sysctl::RunMode::Run, sysctl::PowerState::On);
+                    sysctl::reset(pc, sysctl::Domain::$powerDomain);
+
+                    // Stop Timers
+                    tim.ctl.write(|w|
+                                  w.taen().clear_bit()
+                                  .tben().clear_bit()
+                                  .tastall().set_bit()
+                    );
+
+                    // GPTMCFG = 0x0 (chained - 2x16 = 32bits) This
+                    // will not force 32bits wide timer, this will
+                    // really force the wider range to be used (32 for
+                    // 16/32bits timers, 64 for 32/64).
+                    tim.cfg.write(|w| w.cfg()._32_bit_timer());
+
+                    tim.tamr.write(|w| w.tamr().period());
+
+                    let mut timer = Timer {
+                        tim:tim,
+                        clocks: *clocks,
+                        timeout: Hertz(0),
+                    };
+                    timer.start(timeout);
+
+                    timer
+                }
+
+                /// Starts listening for an `event`
+                pub fn listen(&mut self, event: Event) {
+                    match event {
+                        Event::TimeOut => {
+                            // Enable update event interrupt
+                            self.tim.imr.modify(|_,w|  w.tatoim().set_bit());
+                        }
+                    }
+                }
+
+                /// Stops listening for an `event`
+                pub fn unlisten(&mut self, event: Event) {
+                    match event {
+                        Event::TimeOut => {
+                            // Enable update event interrupt
+                            self.tim.imr.modify(|_,w| w.tatoim().clear_bit());
+                        }
+                    }
+                }
+
+                /// Releases the TIM peripheral
+                pub fn free(self) -> $TIM {
+                    // pause counter
+                    self.tim.ctl.write(|w|
+                                  w.taen().clear_bit()
+                                  .tben().clear_bit());
+                    self.tim
+                }
+            }
+        )+
+    }
+}
+
+hal! {
+    TIMER0: (timer0, Timer0),
+    TIMER1: (timer1, Timer1),
+    TIMER2: (timer2, Timer2),
+    TIMER3: (timer3, Timer3),
+    TIMER4: (timer4, Timer4),
+    TIMER5: (timer5, Timer5),
+
+    WTIMER0: (wtimer0, WideTimer0),
+    WTIMER1: (wtimer1, WideTimer1),
+    WTIMER2: (wtimer2, WideTimer2),
+    WTIMER3: (wtimer3, WideTimer3),
+    WTIMER4: (wtimer4, WideTimer4),
+    WTIMER5: (wtimer5, WideTimer5),
+
+    // TIM3: (tim3, tim3en, tim3rst),
```

Comment:
Can probably delete these lines?

Reply:
Ooops^2 ! Sorry for letting these 1 stupid mistakes in the changes...